src/t_list.c

```diff
@@ -385,7 +385,53 @@ void lsetCommand(client *c) {
     }
 }
 
-void listElementsRemoved(client *c, robj *key, int where, robj *o) {
+/* A helper for replying with a list's range as a multi-bulk. For backward
+* compatibility, start needs to be less than end. The direction affects how
+* elements are iterated. */
+void addListRangeReply(client *c, robj *o, long start, long end, int direction) {
```

Comment:
the `direction` argument is a bit confusing IMHO, maybe add a comment explaining it.
alternatively, if we move the part about responding with empty array when end<start to the LRANGE command, maybe start and end can implicitly define the direction?

Reply:
The conversion of negative indices is generic, so I'd rather leave that in the addListRangeReply. Regrettably, the check for start>end depends on it so I chose to keep it that way. Besides, I prefer explicitness here for some reason.